### PR TITLE
CON-9193 update-csi for k8s 1.27

### PR DESCRIPTION
**Pull request by llDrLove**

### CODE REVIEW

**Review by timoreimann — COMMENTED** [review]
LGTM with one request: would you mind removing unsupported 1.23 from the CI?

**Comment by llDrLove** [comment]
@timoreimann , on it.

File: test/e2e/handle-image.sh [hunk 1]
```diff
@@ -34,7 +34,7 @@ readonly OPERATION="$1"
 
 case "${OPERATION}" in
   build)
-    docker build -t "${IMAGE}" --build-arg KUBE_VERSION_1_26 --build-arg KUBE_VERSION_1_26_E2E_BIN_SHA256_CHECKSUM --build-arg KUBE_VERSION_1_25 --build-arg KUBE_VERSION_1_25_E2E_BIN_SHA256_CHECKSUM --build-arg KUBE_VERSION_1_24 --build-arg KUBE_VERSION_1_24_E2E_BIN_SHA256_CHECKSUM --build-arg KUBE_VERSION_1_23 --build-arg KUBE_VERSION_1_23_E2E_BIN_SHA256_CHECKSUM --build-arg KUBE_VERSION_1_22 --build-arg KUBE_VERSION_1_22_E2E_BIN_SHA256_CHECKSUM --build-arg KUBE_VERSION_1_21 --build-arg KUBE_VERSION_1_21_E2E_BIN_SHA256_CHECKSUM -f "${SCRIPT_DIR}/Dockerfile" "${SCRIPT_DIR}"
```
Comment (timoreimann) [inline test/e2e/handle-image.sh:37]: I think we can drop the 1.23-and-below references here as well.

**Review by timoreimann — COMMENTED** [review]
One more comment.

Please also delete `test/e2e/testdrivers/1.{2,3}`.

File: test/e2e/handle-image.sh (hunk 1 above)
Comment (llDrLove) [reply test/e2e/handle-image.sh:37]: I dropped the 1.23 and below references and dropped the testdrivers/1.22 and 1.23 files respectively.

**Review by timoreimann — APPROVED** [review]

**Pull request merged**